admin/lib/validator/restart-validator.js

```diff
@@ -0,0 +1,18 @@
+const ValidationError = require('../validation-error')
+const restartValidator = {}
+
+/**
+ * Validate restart reason
+ * @param reason
+ * @param specifyReason
+ * @returns {Boolean}
+ */
+restartValidator.validateReason = (reason, specifyReason) => {
+  const validationError = new ValidationError()
+  if (reason === 'Did not complete' && specifyReason.length === 0) {
+    validationError.addError('didNotCompleteInfo', 'Please specify further information when "Did not complete" option is selected')
```

Comment:
This error message should be coming from an external file please.

Reply:
@jon-shipley it is now